include/crm_internal.h

```diff
@@ -22,6 +22,31 @@
 #  include <crm/common/ipcs.h>
 #  include <crm/common/internal.h>
 
+/*
+ * The man pages for both curses and ncurses suggest inclusion of "curses.h".
+ * We believe the following to be acceptable and portable.
+ */
+
+#  if defined(HAVE_LIBNCURSES) || defined(HAVE_LIBCURSES)
+#    if defined(HAVE_NCURSES_H) && !defined(HAVE_INCOMPATIBLE_PRINTW)
+#      include <ncurses.h>
+#      define CURSES_ENABLED 1
+#    elif defined(HAVE_NCURSES_NCURSES_H) && !defined(HAVE_INCOMPATIBLE_PRINTW)
+#      include <ncurses/ncurses.h>
+#      define CURSES_ENABLED 1
+#    elif defined(HAVE_CURSES_H) && !defined(HAVE_INCOMPATIBLE_PRINTW)
+#      include <curses.h>
+#      define CURSES_ENABLED 1
+#    elif defined(HAVE_CURSES_CURSES_H) && !defined(HAVE_INCOMPATIBLE_PRINTW)
+#      include <curses/curses.h>
+#      define CURSES_ENABLED 1
+#    else
+#      define CURSES_ENABLED 0
+#    endif
+#  else
+#    define CURSES_ENABLED 0
+#  endif
+
```

Comment:
Seeing this in practice, moving status_print() to pe_status_private.h was a good idea, but I'm having second thoughts about the include. :-/ Given that the vast majority of files that include crm_internal.h don't link against the curses library, I think we should probably leave curses_internal.h for just this block. Once we're able to drop the now-obsolete print-based functions in libpe_status, crm_mon will be the only thing left with any curses code, so I think it makes sense to keep it separate from everything else.

Reply:
This probably means I can ditch the xml.c portion, too.